service/orchestrator/certificates.go

```diff
@@ -85,12 +86,18 @@ func (svc *Service) ListCertificates(ctx context.Context, req *orchestrator.List
 	}
 
 	// We only list certificates the user is authorized to see (w.r.t. the cloud service)
-	var conds []any
+	var (
+		query []string
+		conds []any
+	)
+
 	all, allowed := svc.authz.AllowedCloudServices(ctx)
 	if !all {
-		conds = append([]any{"cloud_service_id IN ?"}, []any{allowed})
+		query = append(query, "cloud_service_id IN ?")
+		conds = append(conds, allowed)
 	}
 
+	conds = append([]any{strings.Join(query, " AND ")}, conds...)
```

Comment:
If we use the same paradigm of having a query and a conds object, which then later gets combined, it works. We should probably think of having this pattern extracted into a function or similar

Reply:
Here, the `" AND "` part never gets executed/joined because we only have a single query (`"cloud_service_id IN ?"`)?! As a general pattern, yes, I could understand it; but here it is just adding something that's not used. So what do we actually achieve here?

Reply:
We achieve that it works ;)

Reply:
aha

Reply:
I extracted it into a function